ui/tests/acceptance/clients-list-test.js

```diff
@@ -95,6 +95,33 @@ module('Acceptance | clients list', function(hooks) {
 
     assert.equal(ClientsList.nodes[4].state.text, 'draining');
     assert.ok(ClientsList.nodes[4].state.isInfo, 'expected info class');
+
+    await ClientsList.sortBy('state');
+
+    assert.deepEqual(ClientsList.nodes.mapBy('state.text'), [
+      'ready',
+      'initializing',
+      'ineligible',
+      'draining',
+      'down',
+    ]);
+
+    // Simulate a client state change arriving through polling
+    let readyClient = this.owner
+      .lookup('service:store')
+      .peekAll('node')
+      .findBy('modifyIndex', 4);
+    readyClient.set('schedulingEligibility', 'ineligible');
```

Comment:
I don't like this, but as you point out there isn't a good way to emulate blocking queries in tests, sooooooooo....